contexts/tx.tsx

```diff
@@ -0,0 +1,122 @@
+import { createContext, ReactNode, useContext } from 'react'
+import { TxRaw } from 'cosmjs-types/cosmos/tx/v1beta1/tx'
+import { isDeliverTxSuccess } from '@cosmjs/stargate'
+import { coins } from '@cosmjs/stargate'
+import useToaster, { ToastPayload, ToastTypes } from 'hooks/toast'
+import { useMeshClient } from 'client'
+import { useWallet } from '@cosmos-kit/react'
+
+// Context to handle simple signingClient transactions
+export interface Msg {
+  typeUrl: string
+  value: any
+}
+
+export interface TxOptions {
+  gas?: number
+  toast?: {
+    title?: ToastPayload['title']
+    message?: ToastPayload['message']
+    type?: ToastTypes
+    actions?: JSX.Element
+  }
+}
+
+export interface TxContext {
+  tx: (
+    msgs: Msg[],
+    options: TxOptions,
+    callback?: (log: any) => void,
+  ) => Promise<void>
+}
+
+export const Tx = createContext<TxContext>({
+  tx: () => new Promise(() => {}),
+})
+
+export function TxProvider({ children }: { children: ReactNode }) {
```

Comment:
Ahhh.... this is why you build messages, so you can have one submit item here.

Hmmm... I get the idea and feedback, but I still don't like all the indirection around just calling the client.

Reply:
What about some function that would be called:

`await doTx(async () => meshLockupClient.stake({amount: "1000", validator: "juno1swfuihwiufh"}));`

It could set up the toasts and then call the passed function, capturing errors or success and doing the toasts.

I feel this is much cleaner than trying to add all kind of changes to each generation event and then submitting them in some custom way. We can avoid a lot of the hand-written client code, and rather just add a few lines around each action to ensure toasts are shown

Reply:
We don't want to the toasts to show when the message hasn't been broadcast yet.
`MessageComposer` & then `signingCosmWasmClient.broadcastTx` is the only way to do this.

Think of the user flow as following:
```
Press button -> Keplr popup -> Broadcast -> "Broadcasting..." toast -> Success!
```
In my opinion this is optimal & better UX.